### PR TITLE
Dualsim: fallback to default setting before Optional.absent

### PR DESCRIPTION
**Pull request by vdveer**
#371  Built it for personal use, maybe it's handy for others as well.

### CODE REVIEW

**Comment by vdveer** [comment]
(Fixed formatting and squashed into new commit)

File: src/org/smssecure/smssecure/ConversationActivity.java [hunk 1]
```diff
@@ -36,6 +36,7 @@
 import android.support.annotation.NonNull;
 import android.support.v4.view.WindowCompat;
 import android.support.v7.app.AlertDialog;
+import android.telephony.SmsManager;
```
Comment (pR0Ps) [inline src/org/smssecure/smssecure/ConversationActivity.java]: Does this need to be added?

File: src/org/smssecure/smssecure/util/dualsim/SubscriptionManagerCompat.java [hunk 2]
```diff
@@ -33,6 +34,14 @@ public SubscriptionManagerCompat(Context context) {
     }
   }
 
+  public static Optional<Integer> getDefaultMessagingSubscriptionId() {
+    if (Build.VERSION.SDK_INT < 22) {
+      return Optional.absent();
+    }
+
+    return SmsManager.getDefaultSmsSubscriptionId() < 0 ? Optional.<Integer>absent() : Optional.of(SmsManager.getDefaultSmsSubscriptionId());
```
Comment (pR0Ps) [inline src/org/smssecure/smssecure/util/dualsim/SubscriptionManagerCompat.java]: please break into a non-ternary if stmt.

Also, can the `Optional.<Integer>absent()` be replaced with `Optional.absent()` (like the build version guard block)

**Comment by pR0Ps** [comment]
In general this looks good although I can't test it (I don't have access to a dual-SIM phone).

@vdveer: Would you consider submitting this patch to [the upstream project](https://github.com/WhisperSystems/Signal-Android)? Their dual-SIM code is basically the same as ours (meaning we could easily pull it in with the next set of upstream commits) and I'm sure they would appreciate it.

**Comment by vdveer** [comment]
Will do later on!

**Comment by vdveer** [comment]
While testing (while reading about the contract, contribution.md etcetera) I found out that this code is not needed in upstream. The first manual selection of subscriptionID is always done when selecting SMS as a transport option (default is always back to online messaging). So I think it's better to merge this PR here.

**Pull request closed without merging**